client/src/components/SentimentAnalysis/SentimentAnalysis.js

```diff
@@ -72,16 +109,66 @@ const SentimentAnalysis = props => {
    * - If not, show empty textarea so user can filled in themselves
    */
   const displayTextareaValue = label => {
-    if (locationState === undefined) {
+    if (id === undefined) {
       return '';
     }
-    return label === 'headline'
-      ? locationState.ad.headline
-      : locationState.ad.content;
+    return label === 'headline' ? headline.text : content.text;
+  };
+
+  /**
+   * Generates a PNG screenshot of the current result page and
+   * automatically saves it to user's local machine
+   * Note: download as PNG because of its flexibility in dimensions
+   * @returns {Void} doesn't return anything
+   */
+  const downloadScreenshot = () => {
+    setDownloadStatus('downloading...');
+    const input = document.getElementById('analysis-container');
+    html2canvas(input).then(canvas => {
+      const imgData = canvas.toDataURL('image/png');
+      saveAs(imgData, 'tardigrade-analysis.png');
```

Comment:
Any way to uniquely identify different downloads?  If the user uses this feature repeatedly they'll get a bunch of the same filename.